core/page_object.py

```diff
@@ -1,12 +1,39 @@
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.wait import WebDriverWait
+from selenium.webdriver.support import expected_conditions as ec
+
+
 class PageObject:
-    """
-    wait for self.selenium_element (visible)
-    redirect all calls to self.selenium_element
-    """
-    def __init__(self, xpath):
+
+    def __init__(self, driver, xpath):
         self.xpath: str = xpath
         self.selenium_element = None
+        self.driver = driver
+
+    def __getattr__(self, item):
+        self.wait()
+        return getattr(self.selenium_element, item)
+
+
+class FindElement(PageObject):
```

Comment:
both PageObject and FindElement must be inherited from BasePageObject since they are equivalent